### PR TITLE
Migrate listpack.c unit tests to new framework 

### PR DESCRIPTION
**Pull request by haoqixu**
This PR migrates the tests related to listpack into new test framework as part of https://github.com/valkey-io/valkey/issues/428.

### CODE REVIEW

File: src/unit/test_listpack.c [hunk 1]
```diff
@@ -0,0 +1,1437 @@
...
+
+    lp = createIntList();
+    TEST_ASSERT(lpLength(lp) == 6);
+    lpFree(lp);
+
+    return 0;
+}
+
+
+int test_listpackCreateList(int argc, char **argv, int flags) {
```
Comment (enjoy-binbin) [inline src/unit/test_listpack.c:144]: I see there are two line breaks in some places here, can you remove it?

Comment (haoqixu) [reply src/unit/test_listpack.c:144]: No problem. done.

File: src/unit/test_listpack.c [hunk 2]
```diff
@@ -0,0 +1,1395 @@
...
+               usec() - start);
+        lpFree(lp);
+    }
+}
+
+/* simple wrapper to used in functions whose return type is not int */
+static inline int _test_assert(int c) {
+    TEST_ASSERT(c);
+    return 0;
+}
```
Comment (enjoy-binbin) [inline src/unit/test_listpack.c:71]:
```suggestion
}

```

File: src/unit/test_listpack.c [hunk 3]
```diff
@@ -0,0 +1,1395 @@
...
+
+int test_listpackBenchmarkFree(int argc, char **argv, int flags) {
+    UNUSED(argc);
+    UNUSED(argv);
+    UNUSED(flags);
+
+    lpFree(lp);
+    return 0;
+}
```
Comment (enjoy-binbin) [inline src/unit/test_listpack.c:1396]:
```suggestion
}

```

**Review by enjoy-binbin — APPROVED** [review]
seems ok to me. @hwware since you have dealt with this recently, please take another eye.

**Comment by hwware** [comment]
@haoqixu Please do the rebase first, then solve the test case failure if there is. Thanks

**Comment by haoqixu** [comment]
[`Daily / test-ubuntu-jemalloc-fortify`](https://github.com/valkey-io/valkey/actions/runs/10620226673/job/29439586898?pr=949) seems like some kind of false positive, and I find that `unstable` also fails this test: `make all-with-unit-tests CC=gcc OPT=-O3 SERVER_CFLAGS='-Werror -DSERVER_TEST -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3'`. 🤔

**Comment by hwware** [comment]
> [`Daily / test-ubuntu-jemalloc-fortify`](https://github.com/valkey-io/valkey/actions/runs/10620226673/job/29439586898?pr=949) seems like some kind of false positive, and I find that `unstable` also fails this test: `make all-with-unit-tests CC=gcc OPT=-O3 SERVER_CFLAGS='-Werror -DSERVER_TEST -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3'`. 🤔

Thanks, I can not reproduce the problem on machine. I am not sure if gcc 13 is must. I am upgrading it now. If you have time, please take a look, Thanks,

**Review by hwware — APPROVED** [review]
Everything is good. Approved.

**Comment by hwware** [comment]
> > [`Daily / test-ubuntu-jemalloc-fortify`](https://github.com/valkey-io/valkey/actions/runs/10620226673/job/29439586898?pr=949) seems like some kind of false positive, and I find that `unstable` also fails this test: `make all-with-unit-tests CC=gcc OPT=-O3 SERVER_CFLAGS='-Werror -DSERVER_TEST -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3'`. 🤔
> 
> Thanks, I can not reproduce the problem on machine. I am not sure if gcc 13 is must. I am upgrading it now. If you have time, please take a look, Thanks,

I found the problem only exists in gcc13.

**Comment by haoqixu** [comment]
> > > [`Daily / test-ubuntu-jemalloc-fortify`](https://github.com/valkey-io/valkey/actions/runs/10620226673/job/29439586898?pr=949) seems like some kind of false positive, and I find that `unstable` also fails this test: `make all-with-unit-tests CC=gcc OPT=-O3 SERVER_CFLAGS='-Werror -DSERVER_TEST -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3'`. 🤔
> > 
> > 
> > Thanks, I can not reproduce the problem on machine. I am not sure if gcc 13 is must. I am upgrading it now. If you have time, please take a look, Thanks,
> 
> I found the problem only exists in gcc13.

FYI, I run it with gcc 14.2.1

**Comment by zuiderkwast** [comment]
@haoqixu The FORTIFY_SOURCE problem is solved on the unstable branch. Can you merge latest unstable to this branch?

**Comment by enjoy-binbin** [comment]
Merged, thank you for your work.

**Pull request merged**